R/get_cdc_data.R

```diff
@@ -0,0 +1,41 @@
+#' Retrieve CDC growth chart metadataset
+#'
+#' Retrieves CDC growth chart metadataset
+#'
+#' @param type Type of Chart
+#'
+#' A character is expected amongst: "height", "weight", "bmi"
+#'
+#' @return The appropriate growth chart from the CDC data
+#'
+#' @details
+#'
+#' Each data frame will be denoted as below:
+#' \describe{
+#'   \item{\code{SEX}}{double sex where 1 = male, 2 = female}
+#'   \item{\code{AGE}}{double age in months}
+#'   \item{\code{L}}{double Box-Cox transformation for normality}
+#'   \item{\code{M}}{double Median}
+#'   \item{\code{S}}{double Coefficient of variation}
+#' }
+#'
+#' from [cdc websites](https://www.cdc.gov/growthcharts/percentile_data_files.htm)
+#'
+#' @family metadata
+#' @keywords metadata
+#'
+#' @export
+#'
+#' @examples
+#' get_cdc_data(type = "bmi") |> head()
+get_cdc_data <- function(type) {
+  # Limit input values to the three datasets available from CDC data
+  assert_character_scalar(type, values = c("height", "weight", "bmi"))
+
+  # Grab the appropriate one
+  switch(type,
```

Comment:
@rossfarrugia @millerg23 the same script that was used to generate the metadata is still being used but I added an argument to make it internal data, using a R/sysdata.rda file, therefore no longer exports, per r-packages textbook:

![image](https://github.com/pharmaverse/admiralpeds/assets/43018762/d0fec710-0594-433c-877d-106da00559e8)

Reply:
@Pierre-Wallet if you wanna follow along

Reply:
@zdz2101 so you're saying our functions will be able to read in these rda metadata files internally from the package without these datasets being exported to package users right? and package users would be able to generate them themselves using the metadata functions provided or replace with their own versions? sorry Monday brain so just want to be sure i'm grasping this.

Reply:
also let me know if you want me to take a re-review of these. happy to do some testing tomorrow

Reply:
@rossfarrugia exactly what you said!